### ISSUE TITLE
Automatic spell check, custom dictionary OR Integration into mac ecosystem

### ISSUE THREAD

**Issue by ParadoxicalEmboli** [ankitects/anki#1938]
Having a spellchecker on hand is extremely useful. Currently I am using (https://ankiweb.net/shared/info/1410276506) which works with the latest version ( ⁨2.1.54 (b6a7760c)⁩) Qt5 but not Qt6 (for the M1 macs). It would be great to simply have this addon built into anki so that Anki gets spell check support in all fields. Further, this addon can be configured to use any custom dictionary, for me I have a medical, english, and german dictionary plugged in and they work wonderfully. Please integrate spell check into anki.

as an aside, it would also be helpful if anki was able to work with the basic mac system abilities such as look up (hard press) and dictionary/spell checker that is available on the mac itself.

**Comment by glutanimate** [comment]
I have a Qt5/Qt6-compatible version of [Spell Checker](https://github.com/glutanimate/spell-checker-public) in the works that's very close to release. I'd be happy to submit a PR once the new version has seen some testing & maturing as an add-on, if you're still up for that @dae (this time within a reasonable time-frame on my end).

That was the original plan with the add-on, so the code is very Anki-idiomatic and can be easily merged into Anki after some adjustments. With this new release especially, everything is fully typed and in line with Anki's current code style.

Same with the UI, which was written from the ground up with the intent to become part of Anki proper at some point: 

![spell-checker](https://user-images.githubusercontent.com/5459332/176355558-0b3d06eb-52cb-4862-8e82-899be97ec425.png)

And I'd be happy to work on any adjustments, UI or otherwise, as we hit the PR stage of course

**Comment by dae** [comment]
If you're happy to contribute it, that would be appreciated, and I'm sure it would make a lot of Anki users happy.

**Comment by calania** [comment]
@glutanimate Don't want to rush you but have you made any progress since you last posted?

**Comment by glutanimate** [comment]
@calania Yep, on it! Currently in the midst of releasing another major project, but I'm hoping to post a progress update soon.

**Comment by keyb0ardninja** [comment]
Any update on this?

**Comment by user1823** [comment]
@glutanimate, any update on this?

**Comment by Jonhy75** [comment]
Maybe in Anki 23.10?

**Comment by user1823** [comment]
Until the spellchecker gets integrated into Anki, Windows and Linux users can use https://ankiweb.net/shared/info/369581638.

This add-on works with all the latest versions of Anki, including the 23.10 betas.

**Comment by glutanimate** [comment]
Sorry guys, to give you an update: My focus right now in terms of Anki contributions is on helping with native IO. Once that work is complete, I'm planning on revisiting spell checking.

**Comment by noahg1** [comment]
Any update on this?

**Comment by Expertium** [comment]
Bump

**Comment by BlueGreenMagick** [comment]
AnKing also has a [Spell Checker](https://github.com/AnKing-VIP/spell-checker-addon/tree/master) add-on which also allows users to manually add words to the dictionary.
